apps/backend/src/modules/groups/group.entity.ts

```diff
@@ -0,0 +1,86 @@
+import { type Entity } from "~/libs/types/types.js";
+
+import { type GroupCreateResponseDto } from "./libs/types/types.js";
+
+class GroupEntity implements Entity {
+	private id: null | number;
+	private name!: string;
+	private permissionIds!: { id: number }[];
+	private userIds!: { id: number }[];
+
+	private constructor({
+		id,
+		name,
+		permissionIds,
+		userIds,
+	}: {
+		id: null | number;
+		name: string;
+		permissionIds: { id: number }[];
+		userIds: { id: number }[];
+	}) {
+		this.id = id;
+		this.name = name;
+		this.permissionIds = permissionIds;
+		this.userIds = userIds;
+	}
+
+	public static initialize({
+		id,
+		name,
+		permissionIds,
+		userIds,
+	}: {
+		id: number;
+		name: string;
+		permissionIds: { id: number }[];
+		userIds: { id: number }[];
+	}): GroupEntity {
+		return new GroupEntity({
+			id,
+			name,
+			permissionIds,
+			userIds,
+		});
+	}
+
+	public static initializeNew({
+		name,
+		permissionIds,
+		userIds,
+	}: {
+		name: string;
+		permissionIds: { id: number }[];
+		userIds: { id: number }[];
+	}): GroupEntity {
+		return new GroupEntity({
+			id: null,
+			name,
+			permissionIds,
+			userIds,
+		});
+	}
+
+	public toNewObject(): {
+		name: string;
+		permissionIds: { id: number }[];
+		userIds: { id: number }[];
```

Comment:
` { id: number } ` is repeated 12 times. I would create a separate type for this. At least to make it look more attractive

Reply:
I think we just don't need these props in the model, we need permissions and users that will have type Pick<OurCommonUserType, 'id'>

Reply:
Hope I understand correctly
```
permissions: Pick<PermissionModel, "id">[];
users: Pick<UserModel, "id">[];
```

Reply:
and also delete in the group.model.ts

```
public permissions!: { id: number }[];
public users!: { id: number }[];
```

Reply:
yes, rename it to permissions and users instead of permissionIds, we may pass additional fields when creating items.

Reply:
let's use this way if model allows so
> permissions: Pick<PermissionModel, "id">[];